### ISSUE TITLE
NullReferenceException in MultiPartDownload.cs

### ISSUE THREAD

**Issue by Genusatplay** [Toqe/Downloader#22]
Hello.
When initialize download with AlreadyDownloadedRanges with fully downloaded file
get NullReferenceException in SplitToDoRangesForNumberOfParts() - maxRange = null
```
var maxRange = this.ToDoRanges.FirstOrDefault(r => r.Length == this.ToDoRanges.Max(r2 => r2.Length));
this.ToDoRanges.Remove(maxRange);
var range1Start = maxRange.Start;
```

i fix like this
```
private void SplitToDoRangesForNumberOfParts()
{
	try
	{
		while (this.ToDoRanges.Count < this.numberOfParts)
		{
			var maxRange =
				this.ToDoRanges.FirstOrDefault(r => r.Length == this.ToDoRanges.Max(r2 => r2.Length));
			if (maxRange == null)
				return;
			this.ToDoRanges.Remove(maxRange);
			var range1Start = maxRange.Start;
			var range1Length = maxRange.Length / 2;
			var range2Start = range1Start + range1Length;
			var range2Length = maxRange.End - range2Start + 1;
			this.ToDoRanges.Add(new DownloadRange(range1Start, range1Length));
			this.ToDoRanges.Add(new DownloadRange(range2Start, range2Length));
		}
	}
	catch (Exception ex)
	{
		throw ex;
	}
}

private void DetermineFileSizeAndStartDownloads(DownloadCheckResult downloadCheck)
        {
            lock (this.monitor)
            {
                if (AlreadyDownloadedRanges.Count == 1 && AlreadyDownloadedRanges[0].Length == downloadCheck.Size)
                {
                    state = DownloadState.Finished;
                    OnDownloadCompleted(new DownloadEventArgs(this));
                    return;
                }

                this.ToDoRanges = this.DetermineToDoRanges(downloadCheck.Size, this.AlreadyDownloadedRanges);
                this.SplitToDoRangesForNumberOfParts();

                for (int i = 0; i < ToDoRanges.Count; i++)
                {
                    var todoRange = this.ToDoRanges[i];
                    StartDownload(todoRange);
                }
            }
        }

```

**Comment by Toqe** [comment]
Thank you for the hint.
Can you provide a pull request, so I can easily check and integrate your fix?

**Comment by Genusatplay** [comment]
Sure.

**Comment by Genusatplay** [comment]
https://github.com/Toqe/Downloader/pull/23

**Comment by Toqe** [comment]
Fixed with #23, thank you.
